examples/security-token/assets/controller.py

```diff
@@ -0,0 +1,222 @@
+from pyteal import *
+
+def approval_program():
+    """
+    This smart contract acts as a controller of the token. It
+    ensures that the permissions smart contract is called upon each
+    token transfer between non-reserve accounts. The permissions smart contract
+    validates the transfer with some rule checks (eg. asset_amount <= 100).
+    Controller application id is set in clawback-escrow (as a template parameter),
+    to ensure there is a call to controller during a token-transfer.
+    It also handles kill_status of the token. If token is killed then no token transfers
+    are allowed. User can only optOut of the asset.
```

Comment:
except one ?

Reply:
?

Reply:
In spec it is mentioned to kill transfer of asset for everyone except to reserve account, i think that's why we are using this extra variable. right ?

If not why not directly use Asset Freeze feature ?

Reply:
Asset is already frozen, we are using clawback to move assets.
Now if the token is killed, we want to prevent clawback to move token as well.